commit 3e17af253e627d4c652f9a9eae7e02fc1936765b
Author: Dan Murphy <dmurphy@ti.com>
Date:   Fri Jan 11 13:57:07 2019 -0600

    iio: ti-ads8688: Update buffer allocation for timestamps
    
    commit f214ff521fb1f861c8d7f7d0af98b06bf61b3369 upstream.
    
    Per Jonathan Cameron, the buffer needs to allocate room for a
    64 bit timestamp as well as the channels.  Change the buffer
    to allocate this additional space.
    
    Fixes: 2a86487786b5c ("iio: adc: ti-ads8688: add trigger and buffer support")
    Signed-off-by: Dan Murphy <dmurphy@ti.com>
    Cc: <Stable@vger.kernel.org>
    Signed-off-by: Jonathan Cameron <Jonathan.Cameron@huawei.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/iio/adc/ti-ads8688.c b/drivers/iio/adc/ti-ads8688.c
index 184d686ebd99..8b4568edd5cb 100644
--- a/drivers/iio/adc/ti-ads8688.c
+++ b/drivers/iio/adc/ti-ads8688.c
@@ -41,6 +41,7 @@
 
 #define ADS8688_VREF_MV			4096
 #define ADS8688_REALBITS		16
+#define ADS8688_MAX_CHANNELS		8
 
 /*
  * enum ads8688_range - ADS8688 reference voltage range
@@ -385,7 +386,7 @@ static irqreturn_t ads8688_trigger_handler(int irq, void *p)
 {
 	struct iio_poll_func *pf = p;
 	struct iio_dev *indio_dev = pf->indio_dev;
-	u16 buffer[8];
+	u16 buffer[ADS8688_MAX_CHANNELS + sizeof(s64)/sizeof(u16)];
 	int i, j = 0;
 
 	for (i = 0; i < indio_dev->masklength; i++) {
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1316_3e17af253e627d4c652f9a9eae7e02fc1936765b.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1316_3e17af253e627d4c652f9a9eae7e02fc1936765b.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
